test_analysis_all.py

```
@@ -0,0 +1,32 @@
+lib_files = []
+for root, _, files in os.walk(lib_dir):
+    for file in files:
+        if file.endswith(".dart") and not file.endswith(".g.dart") and not file.endswith(".freezed.dart"):
+            lib_files.append(os.path.join(root, file))
+
+# Find all dart test files
+test_files = []
+for root, _, files in os.walk(test_dir):
+    for file in files:
+        if file.endswith("_test.dart"):
+            test_files.append(os.path.join(root, file))
```

Ruff (0.15.7)

[warning] Use `list.extend` to create a transformed list

(PERF401)